### PR TITLE
feat: use eslint-plugin-prefer-arrow-functions

### PR DESCRIPTION
**Pull request by grikomsn**
## Description

This PR integrates [`eslint-plugin-prefer-arrow-functions`](https://github.com/JamieMason/eslint-plugin-prefer-arrow-functions) to autofix arrow function conversion.

## Checklist

- [x] I have made sure the upstream branch for this PR is correct
- [x] I have made sure this PR is ready to merge
- [x] I have made sure that I have assigned reviewers related to this project

## Changes

- [feat(deps): add eslint-plugin-prefer-arrow-functions](https://github.com/strangelove-ventures/style-guide/commit/c0e90878600ee5010472c32ad052a0cd3b10296e)
- [feat(eslint): update stylistic rules](https://github.com/strangelove-ventures/style-guide/commit/b523a9380e0d1efc6913bb726cc91d7ded9af0b6)

## Screenshots

\-

## Testing

\-

## Links/References

- https://github.com/JamieMason/eslint-plugin-prefer-arrow-functions

## Notes

\-

### CODE REVIEW

**Review by joshuanatanielnm — APPROVED** [review]
LGTM

**Comment by grikomsn** [comment]
Waiting for @xanderjl's review since they're a code owner.

**Review by xanderjl — APPROVED** [review]
LGTM 💪

**Pull request merged**
